### PR TITLE
Add unexpectedlengthexception into playback client.

### PR DESCRIPTION
**Pull request by sima-zhu**
This is required for unexpected length exception API working in playback

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/UnexpectedLengthException.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.storage.blob;
+package com.azure.core.implementation;
```
Comment (JonathanGiles) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/UnexpectedLengthException.java]: This should go in the implementation.exception package

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java [hunk 2]
```diff
@@ -12,12 +13,14 @@
...
-    @JsonProperty("Throwable")
-    private Throwable throwable;
-
     @JsonProperty("ClassName")
     private String className;
 
+    @JsonProperty("ErrorMsg")
+    private String errorMsg;
+
+    private Throwable throwable;
```
Comment (alzimmermsft) [inline sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java]: If this isn't being serialized anymore we should remove it.

Comment (sima-zhu) [reply sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java]: Used in default block.

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/UnexpectedLengthException.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/UnexpectedLengthException.java]: Done.

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java [hunk 3]
```diff
@@ -71,4 +78,14 @@ public void throwable(Throwable throwable) {
...
         this.className = className;
     }
+
+    /**
+     * Sets the error message of the class of the throwable. This is used during deserialization the construct the
+     * throwable as the actual class that was thrown.
+     *
+     * @param errorMsg
+     */
+    public void errorMsg(String errorMsg) {
```
[the remark this answers is not in the document]
Comment (sima-zhu) [reply sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java]: Done.

**Comment by sima-zhu** [comment]
Closed it as I merged the changes to https://github.com/Azure/azure-sdk-for-java/pull/5108/

**Pull request closed without merging**